### PR TITLE
Set up base webpack config and install npm dependencies in Dockerfile

### PR DESCRIPTION
**Pull request by antonow**
This PR adds the ability to compile the frontend into static files that can be served by the django backend. It also modifies the Dockerfile to install the frontend project's npm dependencies.

## Description
I began with the vue-django webpack template and updated it to use Vue 2 and Webpack 2. The existing views built by @WGierke's and served by django were integrated into the Vue app in order to be able to leverage the VueRouter for navigating between pages. Rendering one view which initializes a SPA will make page transitions faster than asking the backend for new templates each time a link is clicked and simplifies development because all view rendering and route handling happens in one place.

I had to modify the base image for the dev Dockerfile because I was having difficulty installing node and npm with the Python 3.6 base image. I chose the latest Ubuntu image, but Debian and others would probably work too. I am not sure if there is a need to install the npm dependencies in docker however, because the `npm run build` command which compiles the production assets is run by devs outside of docker anyway, right? Or was the intention to also have Docker compile the production assets if they didn't exist in the dev's repo? If not, I can get rid of these changes to the Dockerfile.

Regarding needing to have 2 servers for frontend development, this would be the case if we wanted to have hot-reloading, but you could also run `npm run build` and serve the compiled assets with only the django backend server running. I do think we should invest the time to get the frontend dev server with hot-reloading working since this will greatly improve the frontend development process, but I was having trouble getting it working after updating Vue and Webpack and wanted to get this PR out.

## Reference to official issue
[Set up a basic npm vue/webpack setup #6](https://github.com/concept-to-clinic/concept-to-clinic/issues/6)

## Motivation and Context
We need a way to compile the frontend app and serve it from django.

## How Has This Been Tested?
I ran `docker-compose -f local.yml build` to ensure that the project's npm dependencies get installed.
I ran `npm install` and `npm run build` from the frontend directory, then `docker-compose -f local.yml up`, and navigated to localhost:8000 and saw this:

screen shot 2017-09-06 at 10 51 49 pm

## CLA
- [x] I have signed the CLA; if other committers are in the commit history, they have signed the CLA as well

### CODE REVIEW

File: interface/requirements/local.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 coverage
 factory_boy==2.8.1
-ipython==6.1.0
+--pre ipython==6.1.0
```
Comment (WGierke) [inline interface/requirements/local.txt]: Hey @antonow cool stuff! But can you tell me why this is desirable?

**Comment by WGierke** [comment]
If you remove the space in front of your 'X' in the PR description, the CLA test should pass :)
Could you adapt the current failing test, that checks whether the 'Open Image' page exists, to the new Vue.js frontend? I'm still not quite sure why the second test fails ...

**Comment by lamby** [comment]
> If you remove the space in front of your 'X' in the PR description, the CLA test should pass :)

No, you need to sign the CLA on https://concepttoclinic.drivendata.org - it's not "just" a text field here!

File: interface/requirements/local.txt (hunk 1 above)
Comment (antonow) [reply interface/requirements/local.txt]: When I changed the docker base image to Ubuntu, `pip` wasn't able to find version 6.1.0 of `ipython`. I found somebody on SO saying that pre-release versions aren't displayed by default so you need to prepend `--pre` to get them.

Comment (antonow) [reply interface/requirements/local.txt]: https://pip.pypa.io/en/stable/reference/pip_install/#install-pre

Comment (antonow) [reply interface/requirements/local.txt]: My guess is that the Python 3.6 base image had the desire version of ipython already? We may be able to go back to using that base image depending on the response to the questions in my PR description.

**Comment by lamby** [comment]
Hey @antonow Just needs a rebase against master now :)

**Comment by antonow** [comment]
@lamby - rebased against master again

**Comment by lamby** [comment]
Many thanks!

**Comment by WGierke** [comment]
Hey @antonow thanks a lot for your contribution! Do you happen to have some time to fix the tests that are currently still failing? Otherwise it might sense to open an issue for that so we can keep the tests passing on the master :)

**Pull request merged**